src/nm-policy.c

```diff
@@ -919,6 +919,49 @@ get_best_ip_config (NMPolicy *self,
 }
 
 static void
+update_mdns(NMPolicy *self, NMDnsManager *dns_mgr, const char *ip_iface)
+{
+	guint i;
+	NMPolicyPrivate *priv = NM_POLICY_GET_PRIVATE (self);
+	gs_free NMSettingsConnection **connections = NULL;
+
+	connections = nm_settings_get_connections_sorted (priv->settings, NULL);
+
+	for (i = 0; connections[i]; i++) {
```

Comment:
this is not right. You iterate over all connection-profiles, without regard whether they are currently active or not.
also, `nm_setting_connection_get_interface_name()` is not required to be present.

I would merge this commit in the previous one. But it seems to me, this needs to be re-thought.

Note that for regular DNS settings, the manager tracks it for all devices at the same time. Meaning, whenever a device's configuration changes, we propagate the change to the NMDnsManager via `nm_dns_manager_add_ip6_config()`. But note that that is not only done during `update_ip6_dns()` (which gets called when a different device gets the best default-route), it happens every time something on the device changed.
On the other hand, your `update_mdns()` iterates over all connections, setting nm_dns_manager_set_mdns() for each ifindex it finds. That seems wrong. Instead, pass the mdns configuration for every device during nm_dns_manager_add_ip6_config()